std-bits/base/src/main/java/org/enso/base/Replacer_Cache.java

```diff
@@ -0,0 +1,39 @@
+package org.enso.base;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Replacer_Cache {
+  // Map from replacement string to replacement vector.
+  private static Map<String, Object> cache = new HashMap<String, Object>();
+
+  private static final int lruSize = 5;
+
+  // Circular buffer containing the most recent cache keys.
+  private static final String lru[] = new String[lruSize];
+
+  // Index into the circular buffer.
+  private static int nextSlot = 0;
+
+  public static void put(String key, Object value) {
+    if (lru[nextSlot] != null) {
+      cache.remove(lru[nextSlot]);
+    }
+    lru[nextSlot] = key;
+    nextSlot = (nextSlot + 1) % lruSize;
+
+    cache.put(key, value);
+  }
+
+  public static Object get(String key) {
```

Comment:
we could have just one method `get_or_add(String key, Function<String, Value> callback)`